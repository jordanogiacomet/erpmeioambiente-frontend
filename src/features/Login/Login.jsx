import React from 'react';
import styled from 'styled-components';
import Form from '../../components/Form/Form';
import { Input } from '../../components/Input';
import Heading from '../../components/Heading/Heading';
import Button from '../../components/Button/Button';
import useLogin from '../../hooks/useLogin';
import { faEnvelope, faLock } from '@fortawesome/free-solid-svg-icons';
import { Checkbox } from '../../components/Checkbox';

const Paragraph = styled.p`
  color: ${({ theme }) => theme.colors.secondary};
`;

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
  align-items: center;
  margin-bottom: 24px;
  @media ${({ theme }) => theme.media.max.lg} {
    flex-direction: column;
    gap: 20px;
    align-items: start;
  }
`;

const Login = () => {
  const { email, setEmail, password, setPassword, rememberMe, setRememberMe, error, loading, handleSubmit } = useLogin();

  return (
    <Form onSubmit={handleSubmit}>
      <Heading as="h2" size="big" colorDark={false}>Login In</Heading>
      <Input.Root>
        <Input.Label htmlFor='email' text='Email'/>
        <Input.Field id='email' placeholder="Enter your email" value={email} onChange={(e) => setEmail(e.target.value)} />
        <Input.Icon icon={faEnvelope}/>
      </Input.Root>
      <Input.Root>
        <Input.Label htmlFor='password' text='Password'/>
        <Input.Field id='password' type="password" placeholder="Enter your password" value={password} onChange={(e) => setPassword(e.target.value)} />
        <Input.Icon icon={faLock}/>
      </Input.Root>
      <Wrapper>
        <Checkbox.Root>
          <Checkbox.Field id="remember-me" checked={rememberMe} onChange={() => setRememberMe(!rememberMe)}/>
          <Checkbox.Label htmlFor="remember-me" text="Remember me"/>
        </Checkbox.Root>
        <Button text="Esqueci minha senha" color="#FFFFFF" variant="forgot-password" />
      </Wrapper>
      <Button text={loading ? "Loading..." : "Login In"} type="submit" disabled={loading} />
      {error && <Paragraph>{error}</Paragraph>}
      <Paragraph>NÃ£o possui uma conta? <Button color="#FFFFFF" variant="forgot-password" text="Registre-se"/></Paragraph>
    </Form>
  );
};

export default Login;