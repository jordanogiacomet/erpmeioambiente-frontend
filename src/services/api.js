import axios from 'axios';

const api = axios.create({
  baseURL: 'https://localhost:7186/',
});

export default api;