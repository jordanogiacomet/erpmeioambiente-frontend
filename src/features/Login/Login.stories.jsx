// Login.stories.jsx
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';
import Login from './Login';
import { theme } from '../../styles/theme';
import { MemoryRouter } from 'react-router-dom';

const Template = (args) => (
  <ThemeProvider theme={theme}>
    <MemoryRouter><Login {...args} /></MemoryRouter>
  </ThemeProvider>
);

export default {
  title: 'Components/Login',
  component: Login,
  decorators: [withKnobs],
  argTypes: {
    email: { control: 'text' },
    password: { control: 'text' },
    rememberMe: { control: 'boolean' },
    loading: { control: 'boolean' },
    error: { control: 'text' },
  },
};

export const Default = Template.bind({});
Default.args = {
  email: '',
  password: '',
  rememberMe: false,
  loading: false,
  error: '',
};

export const Customizable = (args) => (
  <ThemeProvider theme={theme}>
    <MemoryRouter>
      <Login
        email={text('Email', '')}
        password={text('Password', '')}
        rememberMe={boolean('Remember Me', false)}
        loading={boolean('Loading', false)}
        error={text('Error', '')}
        {...args}
      />
    </MemoryRouter>
  </ThemeProvider>
);
