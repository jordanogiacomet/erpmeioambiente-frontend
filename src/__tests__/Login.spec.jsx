/* eslint-disable testing-library/no-wait-for-multiple-assertions */
import React from 'react';
import { screen, fireEvent, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import api from '../services/api'; // Importando a inst√¢ncia configurada
import renderTheme from '../styles/render-theme';
import Login from '../features/Login/login';
import '@testing-library/jest-dom/extend-expect';

jest.mock('../services/api');

describe('<Login />', () => {
  beforeEach(() => {
    jest.spyOn(Storage.prototype, 'setItem');
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should render login form and handles login', async () => {
    // Mock resposta da API
    api.post.mockResolvedValueOnce({ data: { token: 'fake-token' } });

    // Renderiza o componente Login
    renderTheme(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    );

    const emailInput = screen.getByLabelText(/email/i);
    const passwordInput = screen.getByLabelText(/senha/i);
    const loginButton = screen.getByRole('button', { name: /login in/i });

    fireEvent.change(emailInput, { target: { value: 'user@example.com' } });
    fireEvent.change(passwordInput, { target: { value: 'password123' } });

    fireEvent.click(loginButton);

    await waitFor(() => {
      expect(api.post).toHaveBeenCalledWith('api/Auth/login', {
        email: 'user@example.com',
        password: 'password123',
      });
      expect(localStorage.setItem).toHaveBeenCalledWith('token', 'fake-token');
    });
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<MemoryRouter><Login /></MemoryRouter>);
    expect(container).toMatchSnapshot();
  });
});

