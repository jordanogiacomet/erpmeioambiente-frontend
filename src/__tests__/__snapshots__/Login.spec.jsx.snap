// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: clamp(320px, 50%, 600px);
  background-color: #000000;
  border-radius: 30px;
  box-shadow: 0 4px 12px rgba(0,0,0,0.1);
  border: 1px solid #000000;
}

.c0>*:not(:last-child) {
  margin-bottom: 16px;
}

.c2 {
  margin-bottom: 16px;
  width: 100%;
  position: relative;
}

.c3 {
  display: block;
  margin-bottom: 8px;
  color: #FFFFFF;
  font-size: 1.6rem;
  font-weight: 600;
}

.c4 {
  padding: 15px 0px;
  width: 100%;
  border: none;
  font-size: 1.6rem;
  border-bottom: 1px solid #FFFFFF;
  background-color: #000000;
  color: #FFFFFF;
  outline: none;
  transition: border-color 0.3s ease;
}

.c4::placeholder {
  font-weight: 600;
  font-size: 1.6rem;
}

.c5 {
  position: absolute;
  right: 10px;
  top: 50%;
  transform: translateY(-50%);
  color: #FFFFFF;
  pointer-events: none;
}

.c1 {
  color: #FFFFFF;
  margin-bottom: 15px;
  font-size: 3rem;
  text-transform: none;
}

.c6 {
  width: 100%;
  padding: 15px 0px;
  background-color: #FFFFFF;
  border: none;
  border-radius: 10px;
  font-size: 1.6rem;
  font-weight: 700;
  cursor: pointer;
  transition: background-color 0.3s ease;
}

.c6:hover {
  background-color: #d0d0d0;
}

.c6:disabled {
  cursor: not-allowed;
}

<div>
  <form
    class="c0"
  >
    <h2
      class="c1"
    >
      Login In
    </h2>
    <div
      class="c2"
    >
      <label
        class="c3"
        for="email"
      >
        Email
      </label>
      <input
        class="c4"
        id="email"
        placeholder="Digite seu email"
        required=""
        type="email"
        value=""
      />
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-envelope c5"
        data-icon="envelope"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="c2"
    >
      <label
        class="c3"
        for="password"
      >
        Senha
      </label>
      <input
        class="c4"
        id="password"
        maxlength="20"
        minlength="5"
        placeholder="Digite sua senha"
        required=""
        type="password"
        value=""
      />
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-lock c5"
        data-icon="lock"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M144 144l0 48 160 0 0-48c0-44.2-35.8-80-80-80s-80 35.8-80 80zM80 192l0-48C80 64.5 144.5 0 224 0s144 64.5 144 144l0 48 16 0c35.3 0 64 28.7 64 64l0 192c0 35.3-28.7 64-64 64L64 512c-35.3 0-64-28.7-64-64L0 256c0-35.3 28.7-64 64-64l16 0z"
          fill="currentColor"
        />
      </svg>
    </div>
    <button
      aria-label="Login In"
      class="c6"
      type="submit"
    >
      Login In
    </button>
  </form>
</div>
`;
